crates/ironrdp-graphics/src/pointer.rs

```diff
@@ -32,70 +32,122 @@ pub enum PointerError {
     #[error("invalid pointer andMask size. Expected: {expected}, actual: {actual}")]
     InvalidAndMaskSize { expected: usize, actual: usize },
     #[error("not supported pointer bpp: {bpp}")]
-    NotSupportedBpp { bpp: usize },
+    NotSupportedBpp { bpp: u16 },
     #[error(transparent)]
     Pdu(#[from] ironrdp_pdu::PduError),
 }
 
 /// Represents RDP pointer in decoded form (color channels stored as RGBA pre-multiplied values)
 #[derive(Debug)]
 pub struct DecodedPointer {
-    pub width: usize,
-    pub height: usize,
-    pub hot_spot_x: usize,
-    pub hot_spot_y: usize,
+    pub width: u16,
+    pub height: u16,
+    pub hotspot_x: u16,
+    pub hotspot_y: u16,
     pub bitmap_data: Vec<u8>,
 }
 
+/// Pointer bitmap rendering target. Defines properties and format of the decoded bitmap.
+#[derive(Clone, Copy, Debug)]
+pub enum PointerBitmapTarget {
+    /// Software rendering target will produce RGBA bitmaps with premultiplied alpha
+    /// and inverted pixels represented as `[0xFF, 0xFF, 0xFF, 0x00]` color.
+    Software,
+    /// Accelerated rendering target will produce RGBA bitmaps with non-premultiplied alpha.
+    /// Inverted pixels will be rendered checkboard pattern.
+    Accelerated,
+}
+
+impl PointerBitmapTarget {
+    fn premultiply_alpha(self) -> bool {
```

Comment:
Nitpick: the name `should_premultiply_alpha` would make it obvious that no operation is performed by this method.
Thought: I’m not sure how useful these very specific function are over something more generic like `is_software` and `is_accelerated`.